common/src/stack/command/stack/commands/load/json/plugin_host.py

```diff
@@ -0,0 +1,192 @@
+			import_data = self.owner.data['host']
+		else:
+			self.owner.log.info('no host data in json file')
+			return
+
+		self.notify('\n\tLoading host\n')
+		# add each host then assign its various values to it
+		for host in import_data:
+			host_name = host['name']
+			parameters = [
+				host_name,
+				f'box={host["box"]}',
+				f'longname={host["appliancelongname"]}',
+				f'rack={host["rack"]}',
+				f'rank={host["rank"]}'
+			]
+			if host['environment']:
+				parameters.append(f'environment={host["environment"]}')
+			self.owner.try_command('add.host',parameters , f'adding host {host["name"]}', 'exists')
+
+
+			# iterate through each interface for the host and set it
+			for interface in host['interface']:
+				self.owner.try_command('add.host.interface',[ host_name, f'interface={interface["interface"]}'], f'adding interface {interface["interface"]}', 'exists')
+
+
+				# iterate over each key in interface, ignoring 'already exists' warnings
+			for interface in host['interface']:
+				for k, v in interface.items():
+					if v and k != 'interface' and k!= 'alias':
+						parameters = [
+							host_name,
+							f'{k}={v}',
+							f'interface={interface["interface"]}',
+							]
+						self.owner.try_command(f'set.host.interface.{k}', parameters, f'setting {host_name} interface {k}', 'exists')
+
+				# the alias cannot be set, so add it here. There can be multiple
+				for alias in interface['alias']:
+					parameters = [
+						host_name,
+						f'alias={alias["alias"]}',
+						f'interface={interface["interface"]}',
+						]
+					self.owner.try_command('add.host.alias', parameters, f'adding {host_name} alias {alias}', 'exists')
+
+			# iterate through each attr for the host and add it
+			for attr in host['attrs']:
+				attr_name = attr['name']
+				attr_value = attr['value']
+				if not isinstance(attr_value, str):
+					attr_value = ' '.join(attr['value'])
+				attr_shadow = attr['shadow']
+				parameters = [
+					host_name,
+					f'attr={attr_name}',
+					f'value={attr_value}',
+					f'shadow={attr_shadow}'
+				]
+				self.owner.try_command('set.host.attr', parameters, f'setting {host["name"]} attr {attr_name}', 'exists')
+
+			# add firewall rules. If the firewall rule already exists, then remove it and add the one in the json
+			for rule in host['firewall']:
+				parameters = [
+					host_name,
+					f'action={rule["action"]}',
+					f'chain={rule["chain"]}',
+					f'protocol={rule["protocol"]}',
+					f'service={rule["service"]}',
+					f'comment={rule["comment"]}',
+					f'flags={rule["flags"]}',
+					f'network={rule["network"]}',
+					f'output-network={rule["output-network"]}',
+					f'rulename={rule["name"]}',
+					f'table={rule["table"]}',
+				]
+				if rule['flags']:
+					parameters.append(f'flags={rule["flags"]}')
+				if rule['comment']:
+					parameters.append(f'comment={rule["comment"]}')
+				if self.owner.try_command('add.host.firewall', parameters, f'adding host firewall rule {rule["name"]}', 'exists') == 1:
```

Comment:
This pattern is fine, but due to line length and repetitive nature of each line, maybe a comment would help to clarify when it does show up.